static-content-hosting/README.md

```diff
@@ -0,0 +1,136 @@
+
+### Prerequisites
+
+1. Latest Azure CLI installed (must be at least 2.40) locally, or you can perform this from Azure Cloud Shell by clicking below or using devcontainers in GitHub.
+
+   [![Launch Azure Cloud Shell](https://learn.microsoft.com/azure/includes/media/cloud-shell-try-it/launchcloudshell.png)](https://shell.azure.com)
+
+   > 💡 The steps shown here and elsewhere in the example use Bash shell commands. On Windows, locally you can use the Windows Subsystem for Linux to run Bash.
+
+1. An Azure subscription.
+
+   The subscription used in this deployment cannot be a [free account](https://azure.microsoft.com/free); it must be a standard EA, pay-as-you-go, or Visual Studio benefit subscription. This is because the resources deployed here are beyond the quotas of free subscriptions.
+
+   > :warning: The user or service principal initiating the deployment process *must* have the following minimal set of Azure role-based access control (RBAC) roles:
+   >
+   > - [User Access Administrator role](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles#user-access-administrator) is *required* at the subscription level since you'll be performing role assignments to managed identities across various resource groups.
+
+### Steps
+
+1. Clone/download this repo locally.
+
+   ```bash
+   git clone https://github.com/mspnp/cloud-design-patterns.git
+   cd cloud-design-patterns/static-content-hosting
+   ```
+
+1. Log into Azure and create an empty resource group.
+
+   ```azurecli
+   az login
+   az account set -s <Name or ID of subscription>
+   ```
+
+1. Create an empty resource group
+
+   ```bash
+   # [This takes less than one minute to run.]
+   az group create -n rg-hosting-static-content -l eastus2
+   ```
+
+1. Control Plane - Deploy a new Azure storage account
+
+   ```bash
+   STORAGE_ACCOUNT_NAME="stvaletblobs$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | fold -w 7 | head -n 1)"
+
+   # [This takes about one minute to run.]
+   az deployment group create -n deploy-hosting-static-content -f bicep/main.bicep -g rg-hosting-static-content -p storageAccountName=$STORAGE_ACCOUNT_NAME
+   ```
+
+   > :warning: for those using devcontainers and running on arm platform, consider `az bicep uninstall && az bicep install --target-platform linux-arm64`
+
+1. [Control Plane] RBAC your user with Storage Blob Data Contributor. This is required to be able to upload files to the recently created storage account
+
+   ```bash
+   ROLEASSIGNMENT_TO_UPLOAD_CONTENT=$(az role assignment create --role ba92f5b4-2d11-453d-a403-e96b0029c9fe --assignee-principal-type user --assignee-object-id $(az ad signed-in-user show --query 'id' -o tsv) --scope $(az storage account show -g rg-hosting-static-content -n $STORAGE_ACCOUNT_NAME --query 'id' -o tsv) --query 'id' -o tsv)
+   echo ROLEASSIGNMENT_TO_UPLOAD_CONTENT: $ROLEASSIGNMENT_TO_UPLOAD_CONTENT
+   ```
```

Comment:
Consider just making this part of the bicep.  You'll need to pass in $(az ad signed-in-user show --query 'id' -o tsv) as a param.

Reply:
just moved this to the bicep @ckittel,

done | addressed from dae6ef749c7c921338c7ec8943d863e909f95882